syntax = "proto3";

message AccessToken {
  string access_token = 1;
}

message AuthResponse {
  int64 status_code = 1;
  optional string user_id = 2;
  optional string message = 3;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message CredentialsResponse {
  int64 status_code = 1;
  optional LoginData data = 2;
  optional string message = 3;
}

message RegisterRequest {
  string username = 1;
  string password = 2;
  string email = 3;
}

message LoginData {
  string refresh_token = 1;
  string access_token = 2;
}
